homeassistant/components/honeywell/__init__.py

```diff
@@ -109,7 +109,8 @@ async def update(self) -> None:
         retries = 3
         while retries > 0:
             try:
-                await self._hass.async_add_executor_job(self.device.refresh)
+                for device in self.devices:
+                    await self._hass.async_add_executor_job(device.refresh)
```

Comment:
Refresh them all in one executor call instead of many as each
call has a cost.


Something like (untested)....
```python3
def _refresh_devices(self):
    """Refresh each enabled device."""
    for device in self.devices:
        device.refresh()
```


```suggestion
                await self._hass.async_add_executor_job(self._refresh_devices)
```